cypress/page/filling-form.page.ts

```diff
@@ -0,0 +1,133 @@
+import { Hobbies } from "./hobbies-enum.page";
+
+class FillingFormPage {
+    private nameSelector: string
+    private lastNameSelector: string
+    private emailSelector: String
+    private genderSelector: String
+    private dateOfBirthSelector: String
+    private mobileNumberSelector: String
+    private hobbiesSelector: String
+    private currentAddressSelector: String
+    private pageURL: string
+    private submit: string
+    private stateSelector: string
+    private citySelector: string
+
+    constructor(){
+        this.nameSelector = "#firstName"
+        this.lastNameSelector = "#lastName"
+        this.emailSelector = "#userEmail"
+        this.genderSelector = "#genterWrapper .col-md-9"
+        this.dateOfBirthSelector = "#dateOfBirthInput"
+        this.mobileNumberSelector = "#userNumber"
+        this.hobbiesSelector = "#hobbiesWrapper .col-md-9"
+        this.currentAddressSelector = "#currentAddress"
+        this.stateSelector = "#state svg"
+        this.citySelector = "#city svg"
+        this.pageURL = "https://demoqa.com/automation-practice-form"
+        this.submit = "form"
+    }
+
+    public visitPage(): void{
+        cy.visit(this.pageURL)
+    }
+
+    public fillForm(personalInformation: any): void{
+            /* Get the hobbies number value 
+            according to the hobbies passed in the parameters */
+            const hobbiesValues = (
+                Object.keys(Hobbies).filter((val) =>
+                    isNaN(Number(val))) as (keyof typeof Hobbies)[]).map((key, value) => {
+                    if(personalInformation['hobbies'].includes(key))
+                        return value + 1;
+                }).filter(Number);
+            cy.get(this.nameSelector).type(personalInformation['name'])
+            cy.get(this.lastNameSelector).type(personalInformation['lastName'])
+            cy.get(this.emailSelector).type(personalInformation['email'])
+            cy.get(this.genderSelector).find(
+                `[value='${personalInformation['gender']}']`).click({force: true})
+            cy.get(this.dateOfBirthSelector).type(`{selectall} ${personalInformation['dateOfBirth']}`)
+            cy.get(this.mobileNumberSelector).type(personalInformation['mobileNumber'])
+            hobbiesValues.forEach((item) => {
+                cy.get(this.hobbiesSelector).find(`[value='${item}']`).click({force: true})
+            })
```

Comment:
so here, for example, you can rewrite this as:
```suggestion
            personalInformation.hobbies.forEach((item) => {
                cy.get(this.hobbiesSelector).contains(item).click({force: true})
            })
```

And you won't need the `hobbiesValues` object and the related logic to initialize it.